spyder/utils/code_analysis/__init__.py

```diff
@@ -0,0 +1,224 @@
+    # document position.
+    "references": {
+        # Find references can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor allows to highlight different text sections at the same time.
+    # The document highlight request is sent from the client to the server to
+    # resolve a document highlights for a given text document position
+    "documentHighlight": {
+        # Code highlighting can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor supports finding symbols on a document.
+    # The document symbol request is sent from the client to the server to list
+    # all symbols found in a given text document.
+    "documentSymbol": {
+        # Find symbols on document can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor allows to autoformat all the document.
+    # The document formatting request is sent from the server to the client to
+    # format a whole document.
+    "formatting": {
+        # Document formatting can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor can autoformat only a selected region on a document.
+    # The document range formatting request is sent from the client to the
+    # server to format a given range in a document.
+    "rangeFormatting": {
+        # Partial document formatting can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor allows to format a document while an edit is taking place.
+    # The document on type formatting request is sent from the client to the
+    # server to format parts of the document during typing.
+    "onTypeFormatting": {
+        # On-Type formatting can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor has an option to go to a function/class/method definition.
+    # The goto definition request is sent from the client to the server to
+    # resolve the definition location of a symbol at a given text document
+    # position.
+    "definition": {
+        # Go-to-definition can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor can give/highlight refactor tips/solutions.
+    # The code action request is sent from the client to the server to compute
+    # commands for a given text document and range. These commands are
+    # typically code fixes to either fix problems or to beautify/refactor code.
+    "codeAction": {
+        # Code hints can be turned on/off dynamically.
+        "dynamicRegistration": True
+    },
+
+    # Editor can display additional commands/statistics per each line.
+    # The code lens request is sent from the client to the server to compute
+    # code lenses for a given text document.
+    # A code lens represents a command that should be shown along with
+    # source text, like the number of references, a way to run tests, etc.
+    "codeLens": {
```

Comment:
I think Spyder doesn't support code lens, does it?

Reply:
No, although It'll be a great addition, (and integration with git), I'll look forward to implement this after you finish this 😄